decidim-core/app/packs/src/decidim/input_character_counter.js

```diff
@@ -91,10 +91,19 @@ export default class InputCharacterCounter {
 
     if (this.maxCharacters > 0) {
       const remaining = this.maxCharacters - numCharacters;
+      let remainingCharacters = this.$input.data("remaining-characters")
       let message = MESSAGES.charactersLeft.other;
       if (remaining === 1) {
         message = MESSAGES.charactersLeft.one;
       }
+      if (document.querySelector(`textarea[data-remaining-characters="${remainingCharacters}"]`)) {
```

Comment:
Since this functionality is strongly related to the emoji picker component, I suggest moving this functionality to the emoji picker itself.

In the character counter, you could trigger/dispatch an event when the character counter is updated at its `updateStatus` methods. Then in the emoji picker, you can listen to this event and decide whether the emoji picker needs to be hidden or displayed based on the remaining characters (that you can pass as an event property in the remaining characters event).

Reply:
Hello,
Thanks a lot, these change requests made me succeed on what i was hoping to do at first !
I think i did what you ask but feel free to review.
Thanks

Reply:
Thanks @eliegaboriau !

What I actually meant that the logic when to hide the emoji picker would be in emoji picker itself.

So, as you have created the custom event, I would actually trigger such custom event every time the character counter is updated. So something like

```js
this.$input[0].dispatchEvent(
  new CustomEvent("characterCounter", { detail: { remaining: remainingCharacters } })
);
```

And then the emoji picker would listen to these events, as in:

```js
elem.addEventListener("characterCounter", (event) => {
  if (event.detail.remaining < 4) {
    // TODO: Hide the emoji picker
  } else {
    // TODO: Show the emoji picker
  }
});
```

This way, the character counter would not have to know about the existence of emoji picker. In the current solution, character counter still has logic that is related to the emoji picker.

Reply:
Ok, that's clearer !
Thanks a lot for your explanations I'm changing this?